File: 1137. N-th Tribonacci Number/tribonacci.js
/**
 * @param {number} n
 * @return {number}
 */
const tribonacci = (n) => {
  if (n === 0) {
    return 0;
  }

  if (n === 1 || n === 2) {
    return 1;
  }

  const res = Array(n + 4).fill(0);
  res[1] = 1;
  res[2] = 1;

  for (let i = 0; i < n - 2; i++) {
    res[i + 3] = res[i] + res[i + 1] + res[i + 2];
  }

  return res[n];
};


console.log('tribonacci: ', tribonacci(25));

File: 1137. N-th Tribonacci Number/tribonacci.py
class Solution:
  def tribonacci(self, n: int) -> int:
    if n == 0:
      return 0
    
    if n == 1 or  n == 2:
      return 1
  
    res = [0] * (n + 4)
    res[1] = 1
    res[2] = 1
  
    for i in range(0, n - 2):
      res[i + 3] = res[i] + res[i + 1] + res[i + 2]

    return res[n]


solution = Solution()
print('tribonacci: ', solution.tribonacci(4))
